### PR TITLE
[DOC] unify param description

### PR DESCRIPTION
**Pull request by Irallia**
For a more uniform documentation, we have established a few rules that can be read in https://github.com/seqan/iGenVar/wiki/Coding-Guide. I have now adapted the existing documentation for this purpose.

### CODE REVIEW

**Review by SGSSGene — APPROVED** [review]
:cry: "west const"? It is a revolution https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/
It is like using useless Emacs over clearly superior vim!

The "[in, out]" tags are great, and I like the " - " before the description of the variables. Makes them much better to read in code, and they look fine in html:

screen

**Comment by Irallia** [comment]
> 😢 "west const"? It is a revolution https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/
> It is like using useless Emacs over clearly superior vim!

Don't know. @eldariont @joshuak94 what do you say? We can also use west const. I just want to have a general solution ^^

**Comment by joshuak94** [comment]
> > cry "west const"? It is a revolution https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/
> > It is like using useless Emacs over clearly superior vim!
> 
> Don't know. @eldariont @joshuak94 what do you say? We can also use west const. I just want to have a general solution ^^

Hmmmm it seems a bit unnatural... But I like the blog post and what they're trying to accomplish so I'm fine with moving to east const...

**Review by joshuak94 — APPROVED** [review]
Looks good! Nice work with the style choices!

**Comment by eldariont** [comment]
> > 😢 "west const"? It is a revolution https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/
> > It is like using useless Emacs over clearly superior vim!
> 
> Don't know. @eldariont @joshuak94 what do you say? We can also use west const. I just want to have a general solution ^^

I don't have strong feeling about this either. The blog post was very interesting but I guess it would be totally fine either way. If we were an independent project I would pick east const. But if seqan uses west const I would stick to that for consistency.

**Comment by Irallia** [comment]
> > > 😢 "west const"? It is a revolution https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/
> > > It is like using useless Emacs over clearly superior vim!
> > 
> > Don't know. @eldariont @joshuak94 what do you say? We can also use west const. I just want to have a general solution ^^
> 
> I don't have strong feeling about this either. The blog post was very interesting but I guess it would be totally fine either way. If we were an independent project I would pick east const. But if seqan uses west const I would stick to that for consistency.

No, seqan uses east const. So, then let's break the egg at the little end and join the east const revolution. 😎
I checked the seqan3 wiki, let's follow these rules:
- when possible, make variable constexpr or const (in that order)
- always use "east-const", i.e. put const on the right of the type that is modified
- if a variable is constexpr, put the constexpr on the left of the type (west-constexpr)
see https://github.com/seqan/seqan3/wiki/General-formatting#variable

**Comment by Irallia** [comment]
@SGSSGene @joshuak94 @eldariont  As I changed everything const related again, could someone recheck my changes? (last two commits) 😁

File: include/modules/sv_detection_methods/analyze_cigar_method.hpp [hunk 1]
```diff
@@ -29,11 +29,11 @@
  *          For more information see the
  *          ([Map Format Specification](https://github.com/samtools/hts-specs/blob/master/SAMv1.pdf)) page 8.
  */
-void analyze_cigar(const std::string & read_name,
-                   const std::string chromosome,
-                   const int32_t query_start_pos,
+void analyze_cigar(std::string const & read_name,
+                   std::string const chromosome,
+                   int32_t query_start_pos,
```
Comment (eldariont) [inline include/modules/sv_detection_methods/analyze_cigar_method.hpp:34]: - can we pass `chromosome` by reference?
- was it intentional to drop the `const` of `query_start_pos`?

File: test/api/junction_detection_methods_test.cpp [hunk 2]
```diff
@@ -253,10 +253,10 @@ TEST(junction_detection, detect_junctions_in_long_reads_sam_file)
...
     Breakend new_breakend_9 {chromosome_1, pos_read_4, strand::forward, sequence_type::reference};
 
-    const std::string & read_name_1 = "m2257/8161/CCS";
-    const std::string & read_name_2 = "m41327/11677/CCS";
-    const std::string & read_name_3 = "m21263/13017/CCS";
-    const std::string & read_name_4 = "m38637/7161/CCS";
+    std::string const & read_name_1 = "m2257/8161/CCS";
+    std::string const & read_name_2 = "m41327/11677/CCS";
+    std::string const & read_name_3 = "m21263/13017/CCS";
+    std::string const & read_name_4 = "m38637/7161/CCS";
```
Comment (SGSSGene) [inline test/api/junction_detection_methods_test.cpp:260]: These shouldn't be references. (You could make them std::string_view)
```suggestion
    std::string const read_name_1 = "m2257/8161/CCS";
    std::string const read_name_2 = "m41327/11677/CCS";
    std::string const read_name_3 = "m21263/13017/CCS";
    std::string const read_name_4 = "m38637/7161/CCS";

```

**Review by SGSSGene — CHANGES_REQUESTED** [review]
"East const" :metal:

There are some std::string references that shouldn't be.

File: include/modules/sv_detection_methods/analyze_cigar_method.hpp (hunk 1 above)
Comment (Irallia) [reply include/modules/sv_detection_methods/analyze_cigar_method.hpp:34]: No, that was a mistake 😅

**Review by SGSSGene — APPROVED** [review]
nice :+1:

**Pull request merged**